cloudinary-test-common/src/main/java/com/cloudinary/test/AbstractStructuredMetadataTest.java

```diff
@@ -0,0 +1,241 @@
+package com.cloudinary.test;
+
+import com.cloudinary.Api;
+import com.cloudinary.Cloudinary;
+import com.cloudinary.api.ApiResponse;
+import com.cloudinary.api.exceptions.BadRequest;
+import com.cloudinary.metadata.*;
+import com.cloudinary.utils.ObjectUtils;
+import org.junit.*;
+import org.junit.rules.TestName;
+
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.*;
+
+import static com.cloudinary.utils.ObjectUtils.asMap;
+import static org.junit.Assert.*;
+import static org.junit.Assume.assumeNotNull;
+
+public abstract class AbstractStructuredMetadataTest extends MockableTest {
+    private static final String METADATA_UPLOADER_TAG = SDK_TEST_TAG + "_uploader";
+
+    protected Api api;
+    public static final List<String> metadataFieldExternalIds = new ArrayList<String>();
+
+    @BeforeClass
+    public static void setUpClass() throws IOException {
+        Cloudinary cloudinary = new Cloudinary();
+        if (cloudinary.config.apiSecret == null) {
+            System.err.println("Please setup environment for Upload test to run");
+        }
+    }
+
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+        Api api = new Cloudinary().api();
+
+        for (String externalId : metadataFieldExternalIds) {
+            try {
+                api.deleteMetadataField(externalId);
+            } catch (Exception ignored) {
+            }
+        }
+    }
+
+    @Rule
+    public TestName currentTest = new TestName();
+
+    @Before
+    public void setUp() {
+        System.out.println("Running " + this.getClass().getName() + "." + currentTest.getMethodName());
+        this.cloudinary = new Cloudinary();
+        assumeNotNull(cloudinary.config.apiSecret);
+        this.api = cloudinary.api();
+    }
+
+    @Test
+    public void testCreateMetadata() throws Exception {
+        StringMetadataField stringField = newFieldInstance("testCreateMetadata_1");
+        ApiResponse result = addFieldToAccount(stringField);
+        assertNotNull(result);
+        assertEquals(stringField.getLabel(), result.get("label"));
+
+        SetMetadataField setField = createSetField("testCreateMetadata_2");
+        result = cloudinary.api().addMetadataField(setField);
+        assertNotNull(result);
+        assertEquals(setField.getLabel(), result.get("label"));
+    }
+
+    @Test
+    public void testDateFieldBehaviour() throws Exception {
```

Comment:
You're testing the default value validation, not setting a date on an asset

Reply:
Yeah, they both go through the same validation process.
I tried to avoid testing all the rule combinations, combined with both update/explicit and default values - It calls for approximately 60 more tests and it basically tests the server side, not the SDK.
I can add 4-5 more tests to cover a bit more if you believe it's necessary.

Reply:
I'm just saying that the name of the test should reflect what it tests :)
something like testDateFieldDefaultVauleValidation